roles/translator/src/downstream_sv1/downstream.rs

```diff
@@ -104,26 +104,30 @@ impl Downstream {
         // role, or the message is sent upwards to the Bridge for translation into a SV2 message
         // and then sent to the SV2 Upstream role.
         let _socket_reader_task = task::spawn(async move {
+            let mut messages = BufReader::new(&*socket_reader).lines();
```

Comment:
By moving this outside the loop it'll only read from the socket once and then the task will die once it completes processing all the messages.

Reply:
In my tests, the stream from `.lines()` blocks appropriately when calling `next()` until it reads end of line or reads EOF (ie. the underlying reader returns `Ok(0)`) - and correctly awaits the underlying reader when it doesn't immediately have data available. In my tests, `TcpStream` also only returned `Ok(0)` when the connection had been remotely terminated, which makes sense, since it is the EOF value - and I checked the underlying std variation and examples which supported this. When I put it all together it worked correctly in my test harness - let me know if/where you're seeing what you're describing and I'll check it out!

Reply:
In my local test everything worked as expected